pkg/performanceprofile/controller/performanceprofile/hypershift/hypershift.go

```diff
@@ -65,16 +83,93 @@ func NewControlPlaneClient(c client.Client, ns string) *ControlPlaneClientImpl {
 	}
 }
 
-func (ci *ControlPlaneClientImpl) getFromConfigMap(ctx context.Context, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error {
-	cm := &corev1.ConfigMap{}
-	if key.Namespace == metav1.NamespaceNone {
-		key.Namespace = ci.managementClusterNamespaceName
+func (ci *ControlPlaneClientImpl) listFromConfigMaps(ctx context.Context, listObj client.ObjectList, opts ...client.ListOption) error {
+	cmList, err := ci.getCorrespondingConfigMaps(ctx, listObj)
+	if err != nil {
+		return err
 	}
-	err := ci.Client.Get(ctx, key, cm, opts...)
+
+	var objectType reflect.Type
+	switch listObj.(type) {
+	case *performancev2.PerformanceProfileList:
+		objectType = reflect.TypeOf(performancev2.PerformanceProfile{})
+	case *machineconfigv1.KubeletConfigList:
+		objectType = reflect.TypeOf(machineconfigv1.KubeletConfig{})
+	case *machineconfigv1.MachineConfigList:
+		objectType = reflect.TypeOf(machineconfigv1.MachineConfig{})
+	case *tunedv1.TunedList:
+		objectType = reflect.TypeOf(tunedv1.Tuned{})
+	default:
+		return fmt.Errorf("unsupported object list type: %T", listObj)
+	}
+
```

Comment:
At this point you already know what type of list you have, so instead of dealing with reflection, why not casting the type and append into its items directly?

Reply:
We should really avoid reflection unless there's a very strong case for it